rachelnavarro/step/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,113 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+
+  private ArrayList<TimeRange> possibleTimes = new ArrayList<TimeRange>();
+  private int currentTime = 0;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    // Get prescheduled events and data about the meeting request.
+    long meetingLength = request.getDuration();
+    ArrayList<String> meetingAttendees = new ArrayList<String>(request.getAttendees());
+    ArrayList<String> meetingOptionalAttendees = new ArrayList<String>(request.getOptionalAttendees());
+    ArrayList<Event> prescheduledEvents = new ArrayList<Event>(events);
```

Comment:
do you really need to turn all these things into ArrayLists?

Reply:
Fixed! And every call to query now zeroes out possible Times and currentTime @andrewnguyen